### PR TITLE
ci: exclude codeql on yaml

### PR DESCRIPTION
**Pull request by asraa**
Fixes https://github.com/slsa-framework/slsa-github-generator/issues/1047

I used the guidance on https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks to create a no-op job with the same name.

Signed-off-by: Asra Ali <asraa@google.com>

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -14,9 +14,13 @@ name: "CodeQL"
 on:
   push:
     branches: [main, "*"]
+    paths-ignore:
+      - "**.yml"
```
Comment (ianlewis) [inline .github/workflows/codeql-analysis.yml:18]: **/*.yml

**Review by ianlewis — APPROVED** [review]

**Comment by asraa** [comment]
> CodeQL Expected — Waiting for status to be reported

TBD I'll make a skipping action or figure out something else.

**Comment by asraa** [comment]
Updated because GitHub recommends creating a duplicate workflow with the same name and job that runs on the ignored paths and performs a successful noop

File: .github/workflows/codeql-analysis-noop.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
```
Comment (ianlewis) [inline .github/workflows/codeql-analysis-noop.yml:10]: nit: Replace with a short comment on why this file exists.

File: .github/workflows/codeql-analysis-noop.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [main, "*"]
+    paths:
+      - "**/*.yml"
```
Comment (ianlewis) [inline .github/workflows/codeql-analysis-noop.yml:18]: If more paths are added to `paths-ignore` I assume they need to be added here as well? Maybe adding a comment to the other file would be good.

**Review by ianlewis — APPROVED** [review]

**Comment by ianlewis** [comment]
@asraa I removed CodeQL as a required check. After this is merged we can hopefully add "CodeQL / Analyze" to the required checks on the protected branch rules.

File: .github/workflows/codeql-analysis-noop.yml (hunk 2 above)
Comment (ianlewis) [reply .github/workflows/codeql-analysis-noop.yml:10]: Also add a URL to the guidance
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks

Comment (asraa) [reply .github/workflows/codeql-analysis-noop.yml:10]: done!

File: .github/workflows/codeql-analysis-noop.yml (hunk 3 above)
Comment (asraa) [reply .github/workflows/codeql-analysis-noop.yml:18]: Nice, if-this-then-that :)

**Pull request merged**